owncloudApp/src/main/java/com/owncloud/android/presentation/transfers/TransfersAdapter.kt

```diff
@@ -220,13 +231,26 @@ class TransfersAdapter(
         }
     }
 
-    fun setData(transfers: List<OCTransfer>) {
+    fun setData(transfers: List<OCTransfer>, spaces: List<OCSpace>) {
         val transfersGroupedByStatus = transfers.groupBy { it.status }
         val newTransferItemsList = mutableListOf<TransferRecyclerItem>()
         transfersGroupedByStatus.forEach { transferMap ->
             val headerItem = HeaderItem(transferMap.key, transferMap.value.size)
             newTransferItemsList.add(headerItem)
-            val transferItems = transferMap.value.sortedByDescending { it.transferEndTimestamp ?: it.id }.map(::TransferItem)
+            val transferItems = transferMap.value.sortedByDescending { it.transferEndTimestamp ?: it.id }.map { transfer ->
+                var spaceName: String? = null
+                var isFromPersonalSpace = false
+                spaces.firstOrNull { it.id == transfer.spaceId }?.let {
+                    if (it.isPersonal) {
+                        spaceName = personalName
+                        isFromPersonalSpace = true
+                    } // TODO: Add else if for shares space
+                    else {
+                        spaceName = it.name
+                    }
+                }
+                TransferItem(transfer, spaceName, isFromPersonalSpace)
```

Comment:
Regarding the first paragraph, ok, I applied it, but there will be much information of a space that is not used though.

Regarding the second paragraph, good catch! I didn't realise that we can have the same space for another account and be outdated because it was not refreshed with the latest changes 👍